apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarType.kt

```diff
@@ -3,8 +3,6 @@ package com.apollographql.apollo.api
 /**
  * Represents a custom GraphQL scalar type
  */
-interface ScalarType {
+expect interface ScalarType {
   fun typeName(): String
-
-  fun javaType(): Class<*>
```

Comment:
Wonder if we should replace this with `KClass`, the only place where it used is to find default scalar type adapter.
`com.apollographql.apollo.api.ScalarTypeAdapters`

Reply:
I tried but it is not possible. It is because the code generation uses this to implement custom types and KClass would not compile in Java code generation.

If we replace it somehow with KClass, then it can be easily moved to common

Reply:
We can replace this with raw string, I can make codegen changes for that.